sound/soc/codecs/rt1011.c

```diff
@@ -1892,11 +1892,17 @@ static int rt1011_set_tdm_slot(struct snd_soc_dai *dai,
 
 	/* Tx slot configuration */
 	tx_slotnum = hweight_long(tx_mask);
-	first_bit = find_next_bit((unsigned long *)&tx_mask, 32, 0);
-	last_bit = find_last_bit((unsigned long *)&tx_mask, 32);
-	if (tx_slotnum > 2 || (last_bit-first_bit) > 1) {
+	if (tx_slotnum > 2 || tx_slotnum == 0) {
 		ret = -EINVAL;
-		dev_dbg(component->dev, "too many tx slots or tx slot location error\n");
+		dev_err(component->dev, "too many tx slots or zero slot\n");
+		goto _set_tdm_err_;
+	}
+
+	first_bit = __ffs(tx_mask);
+	last_bit = __fls(tx_mask);
+	if ((last_bit - first_bit) > 1) {
```

Comment:
I see, that it was there before too, but since you're at it, please also remove the superfluous parentheses around subtraction.

Reply:
yeah. I tried to keep previous code as much as possible and also wanted to keep parentheses 
 when operator precedence is sometimes confusing. But I think this is good point also, hence removed superfluous parentheses.